From: Stefan Ahlers <stef.ahlers@t-online.de>
Description: Use external libjs-cryptojs package instead of included files.
Last-Update: 2016-02-19
Forwarded: no
--- a/resources.qrc
+++ b/resources.qrc
@@ -129,29 +129,6 @@
         <file>data/www/css/style.css</file>
         <file>data/images/splash.svg</file>
         <file>data/images/splash-unrounded.svg</file>
-        <file>data/js/cryptojs/aes.js</file>
-        <file>data/js/cryptojs/hmac-md5.js</file>
-        <file>data/js/cryptojs/hmac-ripemd160.js</file>
-        <file>data/js/cryptojs/hmac-sha1.js</file>
-        <file>data/js/cryptojs/hmac-sha3.js</file>
-        <file>data/js/cryptojs/hmac-sha224.js</file>
-        <file>data/js/cryptojs/hmac-sha256.js</file>
-        <file>data/js/cryptojs/hmac-sha384.js</file>
-        <file>data/js/cryptojs/hmac-sha512.js</file>
-        <file>data/js/cryptojs/md5.js</file>
-        <file>data/js/cryptojs/pbkdf2.js</file>
-        <file>data/js/cryptojs/rabbit.js</file>
-        <file>data/js/cryptojs/rabbit-legacy.js</file>
-        <file>data/js/cryptojs/rc4.js</file>
-        <file>data/js/cryptojs/ripemd160.js</file>
-        <file>data/js/cryptojs/sha1.js</file>
-        <file>data/js/cryptojs/sha3.js</file>
-        <file>data/js/cryptojs/sha224.js</file>
-        <file>data/js/cryptojs/sha256.js</file>
-        <file>data/js/cryptojs/sha384.js</file>
-        <file>data/js/cryptojs/sha512.js</file>
-        <file>data/js/cryptojs/tripledes.js</file>
-        <file>data/js/cryptojs-core.js</file>
         <file>data/www/index.html</file>
         <file>data/images/beatsmusic.svg</file>
         <file>data/images/ipv6-logo.svg</file>
--- a/src/libtomahawk/resolvers/JSResolver.cpp
+++ b/src/libtomahawk/resolvers/JSResolver.cpp
@@ -211,8 +211,8 @@
 
     // Load CrytoJS
     {
-        d->engine->setScriptPath( "cryptojs-core.js" );
-        QFile jslib( RESPATH "js/cryptojs-core.js" );
+        d->engine->setScriptPath( "core.js" );
+        QFile jslib("/usr/share/javascript/cryptojs/components/core.js" );
         jslib.open( QIODevice::ReadOnly );
         d->engine->mainFrame()->evaluateJavaScript( jslib.readAll() );
         jslib.close();
@@ -220,11 +220,11 @@
     {
         QStringList jsfiles;
         jsfiles << "*.js";
-        QDir cryptojs( RESPATH "js/cryptojs" );
+        QDir cryptojs("/usr/share/javascript/cryptojs/" );
         foreach ( QString jsfile, cryptojs.entryList( jsfiles ) )
         {
-            d->engine->setScriptPath( RESPATH "js/cryptojs/" +  jsfile );
-            QFile jslib(  RESPATH "js/cryptojs/" +  jsfile  );
+            d->engine->setScriptPath("/usr/share/javascript/cryptojs/" +  jsfile );
+            QFile jslib("/usr/share/javascript/cryptojs/" +  jsfile  );
             jslib.open( QIODevice::ReadOnly );
             d->engine->mainFrame()->evaluateJavaScript( jslib.readAll() );
             jslib.close();
